pkg/controller/workloaddispatcher/incrementaldispatcher/incremantaldispatcher.go

```diff
@@ -0,0 +1,188 @@
+	return &IncrementalDispatcherReconciler{
+		client:          c,
+		helper:          helper,
+		clock:           realClock,
+		dispatcherName:  dispatcherName,
+		roundStartTimes: make(map[types.NamespacedName]time.Time),
+	}
+}
+func (r *IncrementalDispatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("op", "nominateWorkers")
+	log.V(3).Info("Nominate Worker Clusters with Incremental Dispatcher")
+
+	if r.dispatcherName != kueueconfig.MultiKueueDispatcherModeIncremental {
+		log.V(3).Info("Not a Incremental Dispatcher, skip the reconciliation", "dispatcherName", r.dispatcherName)
+		return reconcile.Result{}, nil
+	}
+
+	wl := &kueue.Workload{}
+	err := r.client.Get(ctx, req.NamespacedName, wl)
+	if err != nil {
+		log.Error(err, "Failed to retrieve Workload, skip the reconciliation", "namespacedName", req.NamespacedName)
+		if apierrors.IsNotFound(err) {
+			r.deleteRoundStartTime(req.NamespacedName)
+		}
+		return reconcile.Result{}, err
+	}
+
+	if !wl.DeletionTimestamp.IsZero() {
+		log.V(3).Info("Workload is deleted, skip the reconciliation", "workload", klog.KObj(wl))
+		r.deleteRoundStartTime(req.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	mkAc, err := dispatcher.GetMultiKueueAdmissionCheck(ctx, r.client, wl)
+	if err != nil {
+		log.Error(err, "Can not get MultiKueue AdmissionCheckState", "workload", klog.KObj(wl))
+		return reconcile.Result{}, err
+	}
+
+	if mkAc == nil || mkAc.State != kueue.CheckStatePending {
+		log.V(3).Info("AdmissionCheckState is not in Pending, skip the reconciliation", "workload", klog.KObj(wl))
+		return reconcile.Result{}, nil
+	}
+
+	// The workload is already assigned to a cluster, no need to nominate workers.
+	if wl.Status.ClusterName != nil {
+		log.V(3).Info("The workload is already assigned to a cluster, no need to nominate workers", "workload", klog.KObj(wl))
+		return reconcile.Result{}, nil
+	}
+
+	remoteClusters, err := dispatcher.GetRemoteClusters(ctx, r.client, r.helper, wl, mkAc.Name)
+	if err != nil {
+		log.Error(err, "Can not get workload group", "workload", klog.KObj(wl))
+		return reconcile.Result{}, err
+	}
+
+	if workload.IsFinished(wl) || !workload.HasQuotaReservation(wl) {
+		log.V(3).Info("Workload is already finished or has no quota reserved, skip the reconciliation", "workload", klog.KObj(wl))
+		r.deleteRoundStartTime(req.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	return r.nominateWorkers(ctx, wl, remoteClusters, log)
+}
+
+func (r *IncrementalDispatcherReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&kueue.Workload{}).
+		Complete(r)
+}
+
+func (r *IncrementalDispatcherReconciler) nominateWorkers(ctx context.Context, wl *kueue.Workload, remoteClusters sets.Set[string], log logr.Logger) (reconcile.Result, error) {
+	log.V(3).Info("NominateWorkers called", "workload", klog.KObj(wl))
```

Comment:
```suggestion
	log.V(3).Info("NominateWorkers called")
```
I think the workload should be included as a log value.